SCRUM-35: Create public profile components and APIs

umedjankabiri opened the pull request:
### Description

This pull request adds the implementation for public profile functionality, including the following features:

- Creation of UI components such as `PublicProfile`, `PublicProfiles`, `ProfileImages`, and `CreatedDate`.
- Addition of styling updates and consistent changes across components.
- Introduction of new API endpoints for fetching public posts and comments (`publicPostsApi`).
- Creation of type definitions and exports for public profiles and associated components.
- Structuring refactors for consistent imports and updated project structure.

umedjankabiri committed:
SCRUM-35 feat: create User public page with styles

umedjankabiri committed:
SCRUM-35 feat: create Total user page with styles

umedjankabiri committed:
SCRUM-35 feat: create Public page with styles

umedjankabiri committed:
SCRUM-35 refactor: refactor imports and function args

umedjankabiri committed:
SCRUM-35 refactor: Refactor structure of project, add index files, update imports

umedjankabiri committed:
SCRUM-35 refactor: Add public user API, endpoints, types and index file for imports

umedjankabiri committed:
this file gives mistakes, I am not have a time to fix it, I just delete pseudo-class.
file was created by Slava Shedu

umedjankabiri committed:
SCRUM-35 refactor: add dynamic import for authApi, to avoid cyclicity

umedjankabiri committed:
SCRUM-35 refactor: change destination

umedjankabiri committed:
SCRUM-35 refactor: update lib

umedjankabiri committed:
SCRUM-35 refactor: implement styles

umedjankabiri committed:
SCRUM-35 refactor: update import for style variable

umedjankabiri committed:
SCRUM-35 refactor: add borders, display users count with individual digits

umedjankabiri committed:
SCRUM-35 refactor: update styles

umedjankabiri committed:
SCRUM-35 refactor: add sort type

umedjankabiri committed:
SCRUM-35 feat: create publicPostsApi with endpoints for fetching posts and comments

umedjankabiri committed:
SCRUM-35 feat: create type definitions for public posts API

umedjankabiri committed:
SCRUM-35 refactor: update styles

umedjankabiri committed:
SCRUM-35 feat: create index file for imports

umedjankabiri committed:
SCRUM-35 refactor: rename type

umedjankabiri committed:
SCRUM-35 refactor: use baseApi instead of authApi

umedjankabiri committed:
SCRUM-35 refactor: delete unused class,  rename mutation

umedjankabiri committed:
SCRUM-35 refactor: rename mutation

umedjankabiri committed:
SCRUM-35 refactor: rename UserPublicProfile to PublicProfiles, update styles

umedjankabiri committed:
SCRUM-35 refactor: change prop name

umedjankabiri committed:
SCRUM-35 refactor: rename UserPublicProfile to PublicProfiles

umedjankabiri committed:
SCRUM-35 refactor: Add remote image loading configuration for staging assets

umedjankabiri committed:
SCRUM-35 refactor: update styles

umedjankabiri committed:
SCRUM-35 refactor: Update publicPosts API types and queries

umedjankabiri committed:
SCRUM-35 feat: create PublicProfiles for public post profiles

umedjankabiri committed:
SCRUM-35 feat: create PublicProfile component for displaying user profiles

umedjankabiri committed:
SCRUM-35 feat: create ProfileImages component to display user profile images

umedjankabiri committed:
SCRUM-35 feat: create CreatedDate component to display relative timestamps

umedjankabiri committed:
SCRUM-35 feat: create UserPublicProfile component with basic styling

umedjankabiri committed:
SCRUM-35 feat: create exports for public profiles and related components

umedjankabiri committed:
Merge remote-tracking branch 'origin/dev' into pages/SCRUM-35-public-page

# Conflicts:
#	src/store/services/api/auth/authApi.ts

umedjankabiri committed:
SCRUM-35 delete: delete unused files

umedjankabiri committed:
SCRUM-35 refactor: add "use client";

umedjankabiri committed:
SCRUM-35 feat: implement SSR render, create getData function for fetching data from server

umedjankabiri committed:
SCRUM-35 refactor: Add image pagination functionality, implemented styles

umedjankabiri committed:
SCRUM-35 refactor: use `Typography` component

eliseevais commented on next.config.ts:
```diff
@@ -14,11 +15,20 @@ const nextConfig: NextConfig = {
+        hostname: "staging-it-incubator.s3.eu-central-1.amazonaws.com",
+        pathname: "/trainee-instagram-api/Image/**",
+      },
+    ],
```
need to resolve conflict with scrum-34 and scrum 36-37

nikitastas replied:
I think Umedjan's version will be good

eliseevais commented on src/app/login/password-restore/page.tsx:
```diff
@@ -8,125 +8,136 @@ import { Card } from "common/components/card/card";
+        setIsLinkSent(true);
+        setIsModalOpen(true);
+      } catch (error) {
+        console.log(error);
```
Please, remove console.log

eliseevais commented on src/app/login/password-restore/page.tsx:
```diff
@@ -8,125 +8,136 @@ import { Card } from "common/components/card/card";
+    try {
+      await resendPasswordRecovery({ email }).unwrap();
+    } catch (error) {
+      console.log(error);
```
Please, remove console.log

eliseevais commented on src/app/login/password-restore/page.tsx:
```diff
@@ -8,125 +8,136 @@ import { Card } from "common/components/card/card";
+              <Link href={"/login"}>Back to Sign In</Link>
+            </Button>
+            <ReCaptcha
+              sitekey="6LdHxG4qAAAAAPKRxEHrlV5VvLFHIf2BO5NMI8YM"
```
Maybe it will be better to move sitekey to constant? for example, to .env

eliseevais commented on src/app/publicPage/publicProfiles/publicProfile/createdDate/createdDate.module.css:
```diff
@@ -0,0 +1,3 @@
+.date {
```
need to add empty lane in the end of the file

eliseevais commented on src/app/publicPage/totalUsersCount/totalUsersCount.tsx:
```diff
@@ -0,0 +1,27 @@
+"use client";
```
can you add empty lane afte "use client"?
we don't have rule for this case

eliseevais commented on src/app/publicPage/userPublicProfile/userPublicProfile.module.css:
```diff
@@ -0,0 +1,3 @@
+.mainWrapper  {
```
Please, add empty lane in the end of the file

umedjankabiri committed:
SCRUM-35 feat: add 'show more/hide' functionality for profile descriptions

umedjankabiri committed:
SCRUM-35 refactor: update ProfileImages to support collapsed view styling

umedjankabiri committed:
SCRUM-35 feat: add format script to package.json for code formatting

umedjankabiri committed:
SCRUM-35 feat: add expandable description component with styles

umedjankabiri committed:
SCRUM-35 feat: add exports for public profile description component

umedjankabiri committed:
SCRUM-35 feat: add pagination to profile images carousel, update styles

umedjankabiri committed:
Refactor codebase for improved readability and consistency

Removed unnecessary imports, standardized imports/export styles, and replaced redundant `console.log` errors with streamlined error handling. Enhanced code clarity by introducing constants and minor formatting changes.

umedjankabiri committed:
Implement post editing features and dropdown components

Added post editing functionality including modals for updating and deleting posts, as well as new dropdown menu components. Enhanced textarea component to support character limits and validation. Updated styles and stories to reflect these changes.

umedjankabiri committed:
Refactor import path for useUpdatePostMutation

Updated the import path for useUpdatePostMutation to reflect correct module structure under "store/services/api/posts/postsApi". This ensures consistency and avoids potential module

umedjankabiri committed:
Refactor imports and update dependencies

Simplified import structures across components and stories, ensuring consistency and proper default/export usage. Updated "@radix-ui/react-dropdown-menu" to version 2.1.7 and corrected paths for "postsApi" related imports, improving code organization and maintainability.

umedjankabiri committed:
Fix prop naming inconsistencies in ReCaptcha stories

Standardize prop names by changing `sitekey` to `siteKey` and `onVerify` to `onVerifyAction`. This ensures consistency with naming conventions and improves clarity in prop usage.

umedjankabiri committed:
Add NEXT_PUBLIC_SITE_KEY to .env.local

This commit introduces the NEXT_PUBLIC_SITE_KEY variable to the .env.local file. It will be used for client-side functionality, likely related to external service integration. This change ensures proper configuration for local development.

umedjankabiri committed:
Refactor ReCaptcha configuration to use environment variable

Replaced hardcoded ReCaptcha site key with an environment variable for improved security and flexibility. Updated related code and .env.local file to align with the new configuration. Removed unused "onVerifyAction" storybook control.

umedjankabiri committed:
Refactor and add post modals, styling, and utility functions

Deleted unused profile styles and reorganized modal-related components for enhanced UX. Introduced new CSS classes, utility functions for date formatting, and defined types for posts, users, and comments in the store layer to streamline type safety and data management. Added dependencies for Radix and Swiper in `package.json`.
